src/GitVersionCore.Tests/IntegrationTests/DevelopScenarios.cs

```diff
@@ -284,5 +284,50 @@ public void PreviousPreReleaseTagShouldBeRespectedWhenCountingCommits()
 
             fixture.AssertFullSemver("1.0.0-alpha.5");
         }
+
+        [Test]
+        public void WhenPreventIncrementOfMergedBranchVersionIsSetToTrueCommitsSinceVersionSourceShouldNotGoDownWhenMergingReleaseToDevelop()
```

Comment:
```suggestion
        public void WhenPreventIncrementOfMergedBranchVersionIsSetToTrue_AndMergingReleaseToDevelop_CommitsSinceVersionSourceShouldNotDecrement()
```

Reply:
@asbjornu Thanks for the feedback. I would love to enhance the documentation, however, to be honest, I am not so thorough with the use cases that `PreventIncrementOfMergedBranchVersion` targets. Having said that, I am sure constructing the fix for this PR would unearth some of the intended use cases, but please feel free to add your thoughts that you think should be added to the relevant docs.

Reply:
@asbjornu Updated the docs